gatsby-config.js

```diff
@@ -80,6 +92,80 @@ module.exports = {
 		'gatsby-plugin-catch-links',
 		'gatsby-transformer-javascript-frontmatter',
 		'gatsby-plugin-react-helmet',
+		{
+			resolve: `@gatsby-contrib/gatsby-plugin-elasticlunr-search`,
+			options: {
+			  // Fields to index
+			  fields: [`title`, `description`, `memberDescriptions`, `members`],
+			  // How to resolve each field`s value for a supported node type
+			  resolvers: {
+				// For any node of type MarkdownRemark, list how to resolve the fields` values
+				MarkdownRemark: {
+					id: (node, getNode) => {
+						if (isNotEmpty(node.frontmatter.title)) {
+							const slug = node.fields.slug;
+							const newPath = slug.substring(1, slug.length-1);
+							return `${node.frontmatter.title}|${newPath}`;
+						} else {
+							const documentationJsComponentDescriptionNode = getNode(node.parent);
+							const documentationJsNode = getNode(documentationJsComponentDescriptionNode.parent);
+							const slug = documentationJsNode.fields.slug;
+							const newPath = slug.substring(1, slug.length-1);
+							const title = slug.substring(14, slug.length-1);
+							const name = documentationJsNode.name;
+							if (isNotEmpty(name) && name.includes('/')) {
+								compareslug = slug;
+								moduleName = name;
+								return `${title}|${newPath}`;
+							} else {
+								moduleName = "";
+							}
+						}
+					},
+					title: (node) => {
+						if (isNotEmpty(node.frontmatter.title)) {
+							return node.frontmatter.title;
+						} else {
+							if (isNotEmpty(moduleName)) {
+								return moduleName;
+							}
+						}
+					},
+					description: (node) => {
+						if (isNotEmpty(node.frontmatter.title)) {
+							return node.internal.content;
+						} else {
+							if (isNotEmpty(moduleName)) {
+								return node.internal.content;
+							}
+						}
+					},
+					memberDescriptions: (node, getNode, getNodesByType) => {
+						memberContent = "";
+						let memberDescriptionContent = "";
+						if (!isNotEmpty(node.frontmatter.title) && isNotEmpty(moduleName)) {
+							const documentationJSNodes = getNodesByType(`DocumentationJs`);
+							documentationJSNodes.forEach( (jsnode) => {
+								if (isNotEmpty(jsnode.name) && jsnode.fields.slug.includes(compareslug)) {
+									if (isNotEmpty(jsnode.description___NODE) && !jsnode.name.includes('/')) {
+										const descNode = getNode(jsnode.description___NODE);
+										memberContent += jsnode.name + " ";
+										memberDescriptionContent += descNode.internal.content + " ";
+									}
+								}
+							});
+							return memberDescriptionContent;
+						}
```

Comment:
I think duplicated words are combined to memberContent.
For example, core/keymap module, "members" field includes those words.
"keyCode removeAll set keyCode addAll add keyCode name name name is set remove "
Some words are duplicated in members field.
Would it be better to remove this duplicate word or leave it as is?
```suggestion
						memberContent = "";
						const memberNameSet = new Set();
						let memberDescriptionContent = "";
						if (!isNotEmpty(node.frontmatter.title) && isNotEmpty(moduleName)) {
							const documentationJSNodes = getNodesByType(`DocumentationJs`);
							documentationJSNodes.forEach((jsnode) => {
								if (isNotEmpty(jsnode.name) && jsnode.fields.slug.includes(compareslug)) {
									if (isNotEmpty(jsnode.description___NODE) && !jsnode.name.includes('/')) {
										const descNode = getNode(jsnode.description___NODE);
										memberNameSet.add(jsnode.name);
										memberDescriptionContent += descNode.internal.content + " ";
									}
								}
							});
							
							memberNameSet.forEach((nodeName) => {
								memberContent += nodeName + " ";
							});
							return memberDescriptionContent;
						}
```

Reply:
It is better to leave it as it is now, as those words become the key for searching and can go to different page links with the same word.

Reply:
Okay, I understand.